server/utils/datetime_utils.py

```
@@ -0,0 +1,162 @@
+        return dt.astimezone().isoformat()
+    except Exception:
+        return "invalid"
+
+def format_date_diff(date1, date2):
+    """
+    Return difference between two datetimes as 'Xd   HH:MM'.
+    Uses app timezone if datetime is naive.
+    date2 can be None (uses now).
+    """
+    # Get timezone from settings
+    tz_name = get_setting_value("TIMEZONE") or "UTC"
```

Ruff (0.14.3)

`get_setting_value` may be undefined, or defined from star imports

(F405)